easy_phi/auth.py

```diff
@@ -262,24 +267,143 @@ def get(self):
 
     post = get
 
+_user_cache = None
+
+
+class PasswordAuthAPIHandler(tornado.web.RequestHandler):
+    user = None
+    password = None
+
+    # this method is static to be used by check_passwords, which is in turn
+    # static to be used by PasswordAuthLoginHandler without class instantiation
+    @staticmethod
```

Comment:
next three functions work only if file with user list is writable from app. It will not work in default configuration